ouroboros-network/src/Ouroboros/Network/Protocol/BlockFetch/Codec.hs

```diff
@@ -27,7 +31,29 @@ import           Text.Printf
 import           Ouroboros.Network.Block (HeaderHash, Point, Serialised (..))
 import qualified Ouroboros.Network.Block as Block
 import           Ouroboros.Network.Codec
+import           Ouroboros.Network.Driver.Limits
 import           Ouroboros.Network.Protocol.BlockFetch.Type
+import           Ouroboros.Network.Protocol.Limits
+
+-- | Byte Limit.
+byteLimitsBlockFetch :: (bytes -> Word) -> ProtocolSizeLimits (BlockFetch block) bytes
+byteLimitsBlockFetch = ProtocolSizeLimits stateToLimit
+  where
+    stateToLimit :: forall (pr :: PeerRole) (st :: BlockFetch block).
+                    PeerHasAgency pr st -> Word
+    stateToLimit (ClientAgency TokIdle)      = smallByteLimit
+    stateToLimit (ServerAgency TokBusy)      = smallByteLimit
+    stateToLimit (ServerAgency TokStreaming) = largeByteLimit
+
+-- | Time Limits
+timeLimitsBlockFetch :: ProtocolTimeLimits (BlockFetch block)
+timeLimitsBlockFetch = ProtocolTimeLimits stateToLimit
+  where
+    stateToLimit :: forall (pr :: PeerRole) (st :: BlockFetch block).
+                    PeerHasAgency pr st -> Maybe DiffTime
+    stateToLimit (ClientAgency TokIdle)      = waitForever
```

Comment:
I don't think we can put a limit on when a peer is supposed to start fetching blocks from us. We may be mostly downstream from them so they only run chainsync towards us. Right now `cardano-node` uses a config which runs chainsync towards 3 peers but only runs blockfetch towards one peer.

In the p2p era we'd always want to start the receiver side lazily, that is there is no need to spin it up unless the peer has sent at least one Mux SDU to it. The need to waitForever still applies since the protocol can still end up in the idle state after starting.